acquia/cli
# Support use of refresh token for authentication

**Motivation**

Adds support for using a refresh token for authentication. 

**Proposed changes**

When sending requests to Cloud API, ACLI will now authenticate by:
1. Checking for either the ACLI_REFRESH_TOKEN env var to find a refresh token. If found, this will be used for authentication.
2. Fall back to looking for saved credentials in cloud_api.conf.

**Alternatives considered**

Using the access token directly. This seemed like a bad idea since League's OAuth Library already has build in support for checking access token expiry and fetching a new one by using a refresh token.

**Testing steps**


1. Follow the [contribution guide](https://github.com/acquia/cli/blob/master/CONTRIBUTING.md#building-and-testing) to set up your development environment.
2. Clear the kernel cache to pick up new and changed commands: `acli ckc`
3. Get a real refresh token:
* Open Google Chrome Dev Tools
* Change to "network" tab to begin recording requests
* Login to Acquia Cloud
* Immediately, before changing URLs, filter the network tab for the "token" resource and select it
* Select the 'preview' subtab to see find the refresh token
4. Run `export ACLI_REFRESH_TOKEN=[my token]`
5. Run an Acquia CLI command and validate that authentication works, even without creds present. E.g., `./bin/acli api:applications:find` 

**Merge requirements**
